### PR TITLE
Added checkbox

### PR DESCRIPTION
**Pull request by say1bt**

### CODE REVIEW

File: FileUploadControl/src/components/TableView/TableView.tsx [hunk 1]
```diff
@@ -21,12 +30,18 @@ const TableView: React.FC<Props> = ({ items, selectedReq, setSelectedReq }) => {
     setSelectedReq(item);
   };
 
+  const onChange = (e: CheckboxChangeEvent, row: any) => {
```
Comment (tilangapramithvidusara) [inline FileUploadControl/src/components/TableView/TableView.tsx:33]: if this function no need then remove it

File: FileUploadControl/src/components/TableView/TableView.tsx [hunk 2]
```diff
@@ -21,12 +30,18 @@ const TableView: React.FC<Props> = ({ items, selectedReq, setSelectedReq }) => {
...
+  };
+
   return (
     <Spin spinning={loading}>
       <div className="header-container">
         <table id={items.id} className="table">
           <thead className="table-head">
             <tr className="table-row">
+              <th></th>
               {items.rows[0].cells.map((cell: any) => (
```
Comment (tilangapramithvidusara) [inline FileUploadControl/src/components/TableView/TableView.tsx:45]: handle these using "?" operator, and define proper types

File: FileUploadControl/src/utils/contextMenu.utils.ts [hunk 3]
```diff
@@ -335,19 +334,9 @@ export const setAsRequirementClick = (
     const selectedObjArr: Obj[] = [];
     let finalObjArr: Obj[] = [];
     if (selectedNodes.length == 1) {
-      console.log(
-        "findIndexRecursive",
-        findIndexRecursive(contentArr, firstElement?.parentElement?.id)[0]
+      const splitIndex: number = contentArr.findIndex(
+        (item: any) => firstElement?.parentElement?.id === item.id
```
Comment (tilangapramithvidusara) [inline FileUploadControl/src/utils/contextMenu.utils.ts:338]: create this function more common way and get this from common utils

**Review by tilangapramithvidusara — CHANGES_REQUESTED** [review]

File: FileUploadControl/src/components/TableView/TableView.tsx (hunk 1 above)
Comment (say1bt) [reply FileUploadControl/src/components/TableView/TableView.tsx:33]: I will comment this until I get the functionality.

File: FileUploadControl/src/utils/contextMenu.utils.ts (hunk 3 above)
Comment (say1bt) [reply FileUploadControl/src/utils/contextMenu.utils.ts:338]: Pasindu aiye will be revamping this function.

File: FileUploadControl/src/components/TableView/TableView.tsx (hunk 2 above)
Comment (say1bt) [reply FileUploadControl/src/components/TableView/TableView.tsx]: Done

File: FileUploadControl/src/components/TableView/TableView.tsx [hunk 4]
```diff
@@ -13,45 +23,55 @@ const TableView: React.FC<Props> = ({ items, selectedReq, setSelectedReq }) => {
...
+                {" "}
+                <td>
+                  <Checkbox onChange={(e) => onChange(e, row)} />
+                </td>
+                {row?.cells?.map((cell: Cell, index: number) => (
                   <td
                     id={cell.id}
                     key={cell.id}
                     style={
                       row.aiSuggestedRequirement && row.id === selectedReq.id
```
Comment (tilangapramithvidusara) [inline FileUploadControl/src/components/TableView/TableView.tsx:63]: use "?" operator

File: FileUploadControl/src/screens/FileUpload.tsx [hunk 5]
```diff
@@ -57,8 +57,8 @@ const FileUpload: React.FC<{
     } catch (error) {
       console.log("Failed to upload files.", error);
       const response = await getFileContent({
-        id: projectId,
-        // id: "6248c85b-e30d-ef11-989a-6045bd12ae6a",
+        // id: projectId,
+        id: "6248c85b-e30d-ef11-989a-6045bd12ae6a",
```
Comment (tilangapramithvidusara) [inline FileUploadControl/src/screens/FileUpload.tsx:61]: remove hardcoded value an put projectId

**Review by tilangapramithvidusara — CHANGES_REQUESTED** [review]

File: FileUploadControl/src/components/TableView/TableView.tsx (hunk 4 above)
Comment (say1bt) [reply FileUploadControl/src/components/TableView/TableView.tsx]: Done

File: FileUploadControl/src/screens/FileUpload.tsx (hunk 5 above)
Comment (say1bt) [reply FileUploadControl/src/screens/FileUpload.tsx]: Done

**Review by tilangapramithvidusara — APPROVED** [review]

File: FileUploadControl/src/pages/TabView.tsx [hunk 6]
```diff
@@ -114,7 +124,7 @@ const TabView: React.FC<TabViewProps> = ({ projectId, accountId }) => {
 
       setResponseData(response?.data);
 
-      // console.log("response", response);
+      console.log("response", response);
     } catch (err: any) {
       console.log("Error in handleContentSubmit", err);
```
Comment (tilangapramithvidusara) [inline FileUploadControl/src/pages/TabView.tsx:129]: handle this error

**Review by tilangapramithvidusara — CHANGES_REQUESTED** [review]

**Review by tilangapramithvidusara — APPROVED** [review]

**Pull request merged**